site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,455 @@
+
+```bash 
+$ kubectl get packagerepository
+NAME                        AGE   DESCRIPTION
+simple-package-repository   40s   Reconcile succeeded
+```
+
+Now, we can see the list of package metadata's available.
+
+```bash
+$ kubectl get packagemetadatas
+NAME                       DISPLAY NAME                   CATEGORIES     SHORT DESCRIPTION   AGE
+nginx.bitnami.vmware.com   Bitnami Nginx Carvel Package   proxy-server   Proxifying Server   56s
+```
+
+**2. List Packages**:
+We can see the list of packages and their version's available for install.
+
+```bash 
+$ kubectl get packages
+NAME                             PACKAGEMETADATA NAME       VERSION   AGE
+nginx.bitnami.vmware.com.1.0.0   nginx.bitnami.vmware.com   1.0.0     1m29s
+```
+
+As we can see, our published nginx helm package is available for us to install.
+
+**3. Create Service Account**: To install the above package, we need to create `default-ns-sa` service account that give PackageInstall CR privileges to create resources in the default namespace.
+
+```bash
+$ kapp deploy -a default-ns-rbac -f https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/develop/examples/rbac/default-ns.yml -y
+Target cluster 'https://192.168.64.32:8443' (nodes: minikube)
+
+Changes
+
+Namespace  Name                     Kind            Conds.  Age  Op      Op st.  Wait to    Rs  Ri
+default    default-ns-role          Role            -       -    create  -       reconcile  -   -
+^          default-ns-role-binding  RoleBinding     -       -    create  -       reconcile  -   -
+^          default-ns-sa            ServiceAccount  -       -    create  -       reconcile  -   -
+
+Op:      3 create, 0 delete, 0 update, 0 noop, 0 exists
+Wait to: 3 reconcile, 0 delete, 0 noop
+
+12:37:16PM: ---- applying 2 changes [0/3 done] ----
+12:37:16PM: create role/default-ns-role (rbac.authorization.k8s.io/v1) namespace: default
+12:37:16PM: create serviceaccount/default-ns-sa (v1) namespace: default
+12:37:16PM: ---- waiting on 2 changes [0/3 done] ----
+12:37:16PM: ok: reconcile serviceaccount/default-ns-sa (v1) namespace: default
+12:37:16PM: ok: reconcile role/default-ns-role (rbac.authorization.k8s.io/v1) namespace: default
+12:37:16PM: ---- applying 1 changes [2/3 done] ----
+12:37:17PM: create rolebinding/default-ns-role-binding (rbac.authorization.k8s.io/v1) namespace: default
+12:37:17PM: ---- waiting on 1 changes [2/3 done] ----
+12:37:17PM: ok: reconcile rolebinding/default-ns-role-binding (rbac.authorization.k8s.io/v1) namespace: default
+12:37:17PM: ---- applying complete [3/3 done] ----
+12:37:17PM: ---- waiting complete [3/3 done] ----
+
+Succeeded
+```
+
+**4. Install the Package**: To install a Carvel Package, we need to create PackageInstall Kubernetes resource. A Package Install will install the nginx helm package and its underlying resources on a Kubernetes cluster. A `PackageInstall` references a `Package`. Thus, we can create the `PackageInstall` yaml from the `Package`.
```

Comment:
Should "Nginx Helm package" be "Nginx Helm chart"?

What does "Thus, we can create the `PackageInstall` yaml from the `Package`" point towards?
Should it be "Thus, we can create the `PackageInstall` yaml for a `Package`". Should it be omitted since we have already mentioned that PackageInstalls reference Packages

Reply:
I want to highlight that you can create `PackageInstall` from `Package` CR. e.g. seeing what fields are exposed, etc.

Reply:
Are we referring to the values schema being defined?